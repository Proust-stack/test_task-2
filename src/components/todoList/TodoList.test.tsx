import { cleanup, screen } from '@testing-library/react';
import { initialTodos, renderTodoList } from '../../tests/helpers/renderWithRouter';

describe('TodoList component', () => {
  afterEach(cleanup);
  it('TodoList snapshot', () => {
    const view = renderTodoList();
    expect(view).toMatchSnapshot();
  });

  it('header exists', () => {
    renderTodoList();
    expect(screen.getByText(/what/i)).toBeInTheDocument();
  });
  it('initial todo exists', () => {
    renderTodoList();
    initialTodos.forEach(el => {
      const todo = screen.getByText(el.title);
      expect(todo).toBeInTheDocument();
    });
  });

});
