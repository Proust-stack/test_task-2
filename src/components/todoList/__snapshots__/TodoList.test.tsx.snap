// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TodoList component TodoList snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="todo_wrapper"
      >
        <div
          class="header"
        >
          <span
            class="header_title"
          >
            What needs to be done
            <div
              style="transform: rotate(180deg);"
            />
          </span>
        </div>
        <div>
          <div
            class="todo"
          >
            <label>
              <input
                class="checkbox"
                type="checkbox"
              />
              <svg
                aria-hidden="true"
                class="checkbox "
                fill="none"
                viewBox="0 0 15 11"
              >
                <path
                  d="M1 4.5L5 9L14 1"
                  stroke="none"
                  stroke-width="1"
                />
              </svg>
              <span>
                Тестовое задание
              </span>
            </label>
          </div>
          <div
            class="todo completed"
          >
            <label>
              <input
                class="checkbox"
                type="checkbox"
              />
              <svg
                aria-hidden="true"
                class="checkbox checkbox--active"
                fill="none"
                viewBox="0 0 15 11"
              >
                <path
                  d="M1 4.5L5 9L14 1"
                  stroke="green"
                  stroke-width="1"
                />
              </svg>
              <span>
                Прекрасный код
              </span>
            </label>
          </div>
          <div
            class="todo"
          >
            <label>
              <input
                class="checkbox"
                type="checkbox"
              />
              <svg
                aria-hidden="true"
                class="checkbox "
                fill="none"
                viewBox="0 0 15 11"
              >
                <path
                  d="M1 4.5L5 9L14 1"
                  stroke="none"
                  stroke-width="1"
                />
              </svg>
              <span>
                Покрытие тестами
              </span>
            </label>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="todo_wrapper"
    >
      <div
        class="header"
      >
        <span
          class="header_title"
        >
          What needs to be done
          <div
            style="transform: rotate(180deg);"
          />
        </span>
      </div>
      <div>
        <div
          class="todo"
        >
          <label>
            <input
              class="checkbox"
              type="checkbox"
            />
            <svg
              aria-hidden="true"
              class="checkbox "
              fill="none"
              viewBox="0 0 15 11"
            >
              <path
                d="M1 4.5L5 9L14 1"
                stroke="none"
                stroke-width="1"
              />
            </svg>
            <span>
              Тестовое задание
            </span>
          </label>
        </div>
        <div
          class="todo completed"
        >
          <label>
            <input
              class="checkbox"
              type="checkbox"
            />
            <svg
              aria-hidden="true"
              class="checkbox checkbox--active"
              fill="none"
              viewBox="0 0 15 11"
            >
              <path
                d="M1 4.5L5 9L14 1"
                stroke="green"
                stroke-width="1"
              />
            </svg>
            <span>
              Прекрасный код
            </span>
          </label>
        </div>
        <div
          class="todo"
        >
          <label>
            <input
              class="checkbox"
              type="checkbox"
            />
            <svg
              aria-hidden="true"
              class="checkbox "
              fill="none"
              viewBox="0 0 15 11"
            >
              <path
                d="M1 4.5L5 9L14 1"
                stroke="none"
                stroke-width="1"
              />
            </svg>
            <span>
              Покрытие тестами
            </span>
          </label>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
